access/gitlab/webhook_server.go

```diff
@@ -87,7 +88,8 @@ func (s *WebhookServer) processWebhook(rw http.ResponseWriter, r *http.Request,
 		http.Error(rw, "", http.StatusBadRequest)
 		return
 	}
-	if r.Header.Get("X-Gitlab-Token") != s.secret {
+	// the length of the secret token is not particularly confidential, so it's ok to leak it here
+	if subtle.ConstantTimeCompare([]byte(r.Header.Get("X-Gitlab-Token")), []byte(s.secret)) == 0 {
```

Comment:
This copies both strings into `[]byte`s; the only other alternative I could think of would've been to just copy the body of `subtle.ConstantTimeCompare` to have it work on `string`, but it didn't seem worth the subsequent maintenance effort.

Reply:
Seems reasonable.